### PR TITLE
feat(script): Add script to export cvs file with user stats

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- We want to be able to get user stats

## This pull request

- Adds a script that gets users stats (mfa enabled, password match, etc)

## Issue that this pull request solves

Closes: https://mozilla-hub.atlassian.net/browse/FXA-7148

## Checklist

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

### CODE REVIEW

File: packages/fxa-auth-server/test/scripts/fixtures/two_email_passwords.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+a@aa.com:passwordzxcv
+a@aa.com:sssssfsfsdf
+b@bbb.com:asdfas
+b@aa.com:passwordzxcv
+b@aa.com:passwordzxcvv
```
[the remark this answers is not in the document]
Comment (clouserw) [reply packages/fxa-auth-server/test/scripts/fixtures/two_email_passwords.txt:5]: If these don't have the actual string 'password' in them, I feel like we'll avoid some false positive run-arounds on auto-detection scripts.  IMO

File: packages/fxa-auth-server/scripts/check-users.js [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+      };
+      const stat = `${s.email},${s.exists},${sanitizeValue(
+        s.passwordMatch
+      )},${sanitizeValue(s.mfaEnabled)},${sanitizeValue(
+        s.keysChangedAt
+      )},${sanitizeValue(s.profileChangedAt)},${sanitizeValue(
+        s.hasSecondaryEmails
+      )},${sanitizeValue(s.isPrimaryEmailVerified)}`;
+
+      console.log(stat);
```
Comment (chenba) [inline packages/fxa-auth-server/scripts/check-users.js:117]: It doesn't look like we need this?

File: packages/fxa-auth-server/scripts/check-users.js [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+        .readFileSync(path.resolve(this.filename))
+        .toString('utf8');
+
+      if (!input.length) {
+        return [];
+      }
+
+      // Parse the input file CSV style
+      return input.split(/\n/).map((s) => {
+        const delimiter = program.delimiter || ':';
```
Comment (pdehaan) [inline packages/fxa-auth-server/scripts/check-users.js:148]: nit: not sure it matters, but you should be able to specify the default option values in the commander config above on L23 directly like this (per https://www.npmjs.com/package/commander#default-option-value):

```diff
-.option('-d, --delimiter [delimiter]', 'Delimiter for input file')
+.option('-d, --delimiter [delimiter]', 'Delimiter for input file', ':')
```

File: packages/fxa-auth-server/scripts/check-users.js [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        return `${s.email},${s.exists},${sanitizeValue(
+          s.passwordMatch
+        )},${sanitizeValue(s.mfaEnabled)},${sanitizeValue(
+          s.keysChangedAt
+        )},${sanitizeValue(s.profileChangedAt)},${sanitizeValue(
+          s.hasSecondaryEmails
+        )},${sanitizeValue(s.isPrimaryEmailVerified)}`;
+      })
+    );
+    const outputFile = program.output || 'stats.csv';
```
Comment (pdehaan) [inline packages/fxa-auth-server/scripts/check-users.js:198]: similar to my previous nit, you should be able to set this in the commander config on L24 like:
```diff
-.option('-o, --output <filename>', 'Output filename to save results to')
+.option('-o, --output <filename>', 'Output filename to save results to', 'stats.csv')
```

File: packages/fxa-auth-server/scripts/check-users.js [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+
+  saveStats() {
+    const stats = this.stats;
+    const output = [
+      'email,exists,passwordMatch,mfaEnabled,keysChangedAt,profileChangedAt,hasSecondaryEmails,isPrimaryEmailVerified',
+    ];
+
+    output.push(
+      ...stats.map((s) => {
+        return `${s.email},${s.exists},${sanitizeValue(
```
Comment (pdehaan) [inline packages/fxa-auth-server/scripts/check-users.js:189]: Q: does `s.email` need to be escaped at all?
I'm not familiar with the live production data or how likely we are to have an `program.delimiter` (default: `:`) in the _local_ part of an email.
https://en.wikipedia.org/wiki/Email_address (which isn't a spec at all) seems to imply that:
> _"space and special characters `"(),:;<>@[\]` **are** allowed with restrictions (they are only allowed inside a quoted string, as described in the paragraph below, and in that quoted string, any backslash or double-quote must be preceded once by a backslash);"_

Or if we use some library if that handles escaping gotchas for us. I've used https://csv.js.org/ in the past, but admittedly only https://csv.js.org/parse/ 99% of the time and not too familiar w/ https://csv.js.org/stringify/.

File: packages/fxa-auth-server/test/scripts/fixtures/two_email_passwords.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (vbudhram) [reply packages/fxa-auth-server/test/scripts/fixtures/two_email_passwords.txt]: I removed this and added basic test cases

File: packages/fxa-auth-server/scripts/check-users.js (hunk 5 above)
Comment (vbudhram) [reply packages/fxa-auth-server/scripts/check-users.js:191]: I thought about a dedicated library to do csv, but then was like meh to add another dependency. Production disallows the local part of an email to contain special characters. I think this is ok.

https://github.com/mozilla/fxa/blob/e06fcbeffab3af3e78be2fa6b8efa36bee7bb2f5/packages/fxa-auth-server/lib/routes/validators.js#L170

**Review by chenba — APPROVED** [review]
Thanks for creating this script 👍

**Pull request merged**